// unchained-ui-react/src/components/controls/Search/SearchCategory.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
  useKeyOnly,
} from '../../../lib';

const SearchCategory = (props) => {
  const {
    active,
    children,
    className,
    renderer
  } = props;

  const classes = cx(
    useKeyOnly(active, 'active'),
    'category',
    className
  );

  const rest = getUnhandledProps(SearchCategory, props);
  const ElementType = getElementType(SearchCategory, props);

  return (
    <ElementType {...rest} className={classes}>
      <div className={'name'}>
        {renderer(props)}
      </div>
      {children}
    </ElementType>
  );
};

SearchCategory._meta = {
  name: 'SearchCategory',
  parent: 'Search',
  type: META.TYPES.CONTROL,
};

SearchCategory.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** The item currently selected by keyboard shortcut. */
  active: PropTypes.bool,

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Display name. */
  name: PropTypes.string, // eslint-disable-line react/no-unused-prop-types

  /**
   * Renders the category contents.
   *
   * @param {object} props - The SearchCategory props object.
   * @returns {*} - Renderable category contents.
   */
  renderer: PropTypes.func,

  /** Array of Search.Result props. */
  results: PropTypes.array, // eslint-disable-line react/no-unused-prop-types
};

SearchCategory.defaultProps = {
  renderer: ({ name }) => name,
};

export default SearchCategory;
