// unchained-ui-react/src/components/containers/List/ListList.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
  useKeyOnly,
} from '../../../lib';

/**
 * A list can contain a sub list.
 */
const ListList = (props) => {
  const {
    children,
    className
  } = props;

  const rest = getUnhandledProps(ListList, props);
  const ElementType = getElementType(ListList, props);

  const classes = cx(
    useKeyOnly(ElementType !== 'ul' && ElementType !== 'ol', 'list'),
    className,
  );

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

ListList._meta = {
  name: 'ListList',
  parent: 'List',
  type: META.TYPES.CONTAINER,
};

ListList.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,
};

export default ListList;
