// unchained-ui-react/src/components/containers/List/ListHeader.jsx
import cx from 'classnames';
import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import {
  createShorthandFactory,
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * A list item can contain a header.
 */
const ListHeader = (props) => {
  const {
    children,
    className,
    content
  } = props;

  const classes = cx(
    'header',
    className
  );

  const rest = getUnhandledProps(ListHeader, props);
  const ElementType = getElementType(ListHeader, props);

  return (
    <ElementType {...rest} className={classes}>
      {_.isNil(children) ? content : children}
    </ElementType>
  );
};

ListHeader._meta = {
  name: 'ListHeader',
  parent: 'List',
  type: META.TYPES.CONTAINER,
};

ListHeader.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Shorthand for primary content. */
  content: customPropTypes.contentShorthand,
};

ListHeader.create = createShorthandFactory(ListHeader, content => ({ content }));

export default ListHeader;
