// unchained-ui-react/src/components/media/Popup/PopupHeader.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  createShorthandFactory,
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * A PopupHeader displays a header in a Popover.
 */
const PopupHeader = (props) => {
  const {
    children,
    className
  } = props;

  const classes = cx(
    'header',
    className
  );

  const rest = getUnhandledProps(PopupHeader, props);
  const ElementType = getElementType(PopupHeader, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

PopupHeader.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,
};

PopupHeader._meta = {
  name: 'PopupHeader',
  type: META.TYPES.MEDIA,
  parent: 'Popup',
};

PopupHeader.create = createShorthandFactory(PopupHeader, children => ({ children }));

export default PopupHeader;
