// unchained-ui-react/src/components/layout/Grid/GridColumn.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
  UV5UI,
  useKeyOnly,
  useOnlyProp,
  useTextAlignProp,
  useValueAndKey,
  useVerticalAlignProp,
  useWidthProp,
} from '../../../lib';

/**
 * A column sub-component for Grid.
 */
const GridColumn = (props) => {
  const {
    children,
    className,
    computer,
    color,
    floated,
    largeScreen,
    mobile,
    only,
    stretched,
    tablet,
    textAlign,
    verticalAlign,
    widescreen,
    width,
  } = props;

  const classes = cx(
    color,
    useKeyOnly(stretched, 'stretched'),
    useOnlyProp(only, 'only'),
    useTextAlignProp(textAlign),
    useValueAndKey(floated, 'floated'),
    useVerticalAlignProp(verticalAlign),
    useWidthProp(computer, 'wide computer'),
    useWidthProp(largeScreen, 'wide large screen'),
    useWidthProp(mobile, 'wide mobile'),
    useWidthProp(tablet, 'wide tablet'),
    useWidthProp(widescreen, 'wide widescreen'),
    useWidthProp(width, 'wide'),
    'column',
    className,
  );

  const rest = getUnhandledProps(GridColumn, props);
  const ElementType = getElementType(GridColumn, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

GridColumn._meta = {
  name: 'GridColumn',
  parent: 'Grid',
  type: META.TYPES.LAYOUT,
};

GridColumn.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** A grid column can be colored. */
  color: PropTypes.oneOf(UV5UI.COLORS),

  /** A column can specify a width for a computer. */
  computer: PropTypes.oneOf(UV5UI.WIDTHS),

  /** A column can sit flush against the left or right edge of a row. */
  floated: PropTypes.oneOf(UV5UI.FLOATS),

  /** A column can specify a width for a large screen device. */
  largeScreen: PropTypes.oneOf(UV5UI.WIDTHS),

  /** A column can specify a width for a mobile device. */
  mobile: PropTypes.oneOf(UV5UI.WIDTHS),

  /** A row can appear only for a specific device, or screen sizes. */
  only: customPropTypes.onlyProp(UV5UI.VISIBILITY),

  /** A column can stretch its contents to take up the entire grid or row height. */
  stretched: PropTypes.bool,

  /** A column can specify a width for a tablet device. */
  tablet: PropTypes.oneOf(UV5UI.WIDTHS),

  /** A column can specify its text alignment. */
  textAlign: PropTypes.oneOf(UV5UI.TEXT_ALIGNMENTS),

  /** A column can specify its vertical alignment to have all its columns vertically centered. */
  verticalAlign: PropTypes.oneOf(UV5UI.VERTICAL_ALIGNMENTS),

  /** A column can specify a width for a wide screen device. */
  widescreen: PropTypes.oneOf(UV5UI.WIDTHS),

  /** Represents width of column. */
  width: PropTypes.oneOf(UV5UI.WIDTHS),
};

export default GridColumn;
