// unchained-ui-react/src/components/media/Image/ImageGroup.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
  UV5UI,
} from '../../../lib';

/**
 * A group of images.
 */
const ImageGroup = (props) => {
  const {
    children,
    className,
    size
  } = props;

  const classes = cx(
    'ui',
    size,
    className,
    'images'
  );

  const rest = getUnhandledProps(ImageGroup, props);
  const ElementType = getElementType(ImageGroup, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

ImageGroup._meta = {
  name: 'ImageGroup',
  parent: 'Image',
  type: META.TYPES.MEDIA,
};

ImageGroup.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** A group of images can be formatted to have the same size. */
  size: PropTypes.oneOf(UV5UI.SIZES),
};

export default ImageGroup;
