// unchained-ui-react/src/components/containers/Table/TableHeader.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
  useKeyOnly,
} from '../../../lib';

/**
 * A table can have a header.
 */
const TableHeader = (props) => {
  const {
    children,
    className,
    fullWidth
  } = props;

  const classes = cx(
    useKeyOnly(fullWidth, 'full-width'),
    className
  );

  const rest = getUnhandledProps(TableHeader, props);
  const ElementType = getElementType(TableHeader, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

TableHeader._meta = {
  name: 'TableHeader',
  type: META.TYPES.CONTAINER,
  parent: 'Table',
};

TableHeader.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** A definition table can have a full width header or footer, filling in the gap left by the first column. */
  fullWidth: PropTypes.bool,
};

TableHeader.defaultProps = {
  as: 'thead',
};

export default TableHeader;
