// unchained-ui-react/src/components/containers/Dropdown/DropdownHeader.jsx
import cx from 'classnames';
import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import {
  createShorthandFactory,
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

import Icon from '../../media/Icon';

/**
 * A dropdown menu can contain a header.
 */
const DropdownHeader = (props) => {
  const {
    children,
    className,
    content,
    icon,
  } = props;

  const classes = cx(
    'header',
    className
  );

  const rest = getUnhandledProps(DropdownHeader, props);
  const ElementType = getElementType(DropdownHeader, props);

  if (!_.isNil(children)) return <ElementType {...rest} className={classes}>{children}</ElementType>;

  return (
    <ElementType {...rest} className={classes}>
      {Icon.create(icon)}
      {content}
    </ElementType>
  );
};

DropdownHeader._meta = {
  name: 'DropdownHeader',
  parent: 'Dropdown',
  type: META.TYPES.CONTAINER,
};

DropdownHeader.propTypes = {
  /** An element type to render as (string or function) */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Shorthand for primary content. */
  content: customPropTypes.contentShorthand,

  /** Shorthand for Icon. */
  icon: customPropTypes.itemShorthand,
};

DropdownHeader.create = createShorthandFactory(DropdownHeader, content => ({ content }));

export default DropdownHeader;
