// unchained-ui-react/webpack.config.js
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const devServer = require('@webpack-blocks/dev-server2');
const splitVendor = require('webpack-blocks-split-vendor');
const happypack = require('webpack-blocks-happypack');
const CompressionPlugin = require('compression-webpack-plugin');
const ExtractTextPlugin = require('extract-text-webpack-plugin');
const StyleLintPlugin = require('stylelint-webpack-plugin');

const {
  addPlugins,
  createConfig,
  entryPoint,
  env,
  setOutput,
  sourceMaps,
  defineConstants,
  webpack,
  resolveAliases,
} = require('@webpack-blocks/webpack2');

const host = process.env.HOST || '0.0.0.0';
const port = process.env.PORT || 3000;
const publicPath = `/${process.env.PUBLIC_PATH || ''}/`.replace('//', '/');
const sourcePath = path.join(process.cwd(), 'docs/app');
const outputPath = path.join(process.cwd(), 'dist');

const babel = () => () => ({
  module: {
    rules: [
      {
        enforce: 'pre',
        test: /\.jsx?$/,
        exclude: /node_modules/,
        use: [
          'eslint-loader',
        ]
      },
      {
        test: /\.jsx?$/,
        exclude: /node_modules/,
        use: [
          'babel-loader',
        ],
      },
    ],
  },
});

const sass = () => () => ({
  module: {
    rules: [
      {
        test: /\.scss$/,
        use: ExtractTextPlugin.extract({
          fallback: 'style-loader',
          use: [
            {
              loader: 'css-loader',
              options: {
                sourceMap: process.env.NODE_ENV !== 'production',
              },
            },
            {
              loader: 'sass-loader',
              options: {
                sourceMap: process.env.NODE_ENV !== 'production',
                precision: 8,
              },
            },
          ],
        }),
      }
    ],
  },
});

const config = createConfig([
  entryPoint({
    app: sourcePath,
  }),
  setOutput({
    filename: 'js/[name].[hash].js',
    path: outputPath,
    publicPath,
  }),
  defineConstants({
    'process.env.NODE_ENV': process.env.NODE_ENV,
    'process.env.PUBLIC_PATH': publicPath,
  }),

  addPlugins([
    new HtmlWebpackPlugin({
      filename: 'index.html',
      minify: {
        collapseactivityBooleanAttributes: true,
        collapseWhitespace: true,
        decodeEntities: true,
        html5: true,
        minifyJS: true,
        processConditionalComments: true,
        removeAttributeQuotes: true,
        removeComments: true,
        removeEmptyAttributes: true,
        removeOptionalTags: true,
        removeRedundantAttributes: true,
        removeScriptTypeAttributes: true,
        removeStyleLinkTypeAttributes: true,
        trimCustomFragments: true,
        useShortDoctype: true,
      },
      template: path.join(process.cwd(), 'public/index.html'),
      title: 'GALE Unchained UI',
      version: require('./package.json').version,
    }),
    new ExtractTextPlugin('css/[name].[hash].css'),
  ]),

  resolveAliases({
    'unchained-ui-react': path.resolve(__dirname, './src/components/index.js'),
  }),

  () => ({
    resolve: {
      modules: ['docs/app', 'node_modules', './'],
    },
    module: {
      rules: [
        {
          test: /\.png$/,
          use: [
            {
              loader: 'url-loader',
              options: {
                prefix: 'images/',
                limit: 8000,
                mimetype: 'image/png',
                name: 'images/[name].[hash].[ext]',
              },
            },
          ],
        },
        {
          test: /(\.jpg|\.jpeg)$/,
          use: [
            {
              loader: 'url-loader',
              options: {
                prefix: 'images/',
                limit: 8000,
                mimetype: 'image/jpeg',
                name: 'images/[name].[hash].[ext]',
              },
            },
          ],
        },
        {
          test: /\.svg$/,
          use: [
            {
              loader: 'url-loader',
              options: {
                prefix: 'fonts/',
                limit: 8000,
                mimetype: 'image/svg+xml',
                name: 'fonts/[name].[hash].[ext]',
              },
            },
          ],
        },
        {
          test: /\.woff$/,
          use: [
            {
              loader: 'url-loader',
              options: {
                prefix: 'fonts/',
                limit: 8000,
                mimetype: 'application/font-woff',
                name: 'fonts/[name].[hash].[ext]',
              },
            },
          ],
        },
        {
          test: /\.woff2$/,
          use: [
            {
              loader: 'url-loader',
              options: {
                prefix: 'fonts/',
                limit: 8000,
                mimetype: 'application/font-woff2',
                name: 'fonts/[name].[hash].[ext]',
              },
            },
          ],
        },
        {
          test: /\.ttf$/,
          use: [
            {
              loader: 'url-loader',
              options: {
                prefix: 'fonts/',
                limit: 8000,
                mimetype: 'application/font-ttf',
                name: 'fonts/[name].[hash].[ext]',
              },
            },
          ],
        },
        {
          test: /\.eot$/,
          use: [
            {
              loader: 'url-loader',
              options: {
                prefix: 'fonts/',
                limit: 8000,
                mimetype: 'application/font-eot',
                name: 'fonts/[name].[hash].[ext]',
              },
            },
          ],
        },
      ],
    },
  }),

  sass(),

  env('development', [
    happypack([
      babel(),
    ]),
    devServer({
      contentBase: 'public',
      stats: 'errors-only',
      publicPath,
      host,
      port,
    }),
    sourceMaps(),
    addPlugins([
      new webpack.NamedModulesPlugin(),
      new StyleLintPlugin(),
    ]),
  ]),

  env('production', [
    happypack([
      babel(),
    ]),
    splitVendor(),
    addPlugins([
      new webpack.optimize.UglifyJsPlugin({
        compress: {
          warnings: false,
          drop_console: true,
          drop_debugger: true,
          dead_code: true,
          unused: true,
        },
        output: {
          comments: false,
        },
      }),
      new CompressionPlugin({
        asset: '[path].gz[query]',
        algorithm: 'gzip',
        test: /\.js$|\.css$|\.html$/,
        threshold: 10240,
        minRatio: 0.8,
      }),
    ]),
  ]),
]);

module.exports = config;
