// unchained-ui-react/src/components/text/Heading/Heading.jsx
import _ from 'lodash';
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
  UV5UI,
  useValueAndKey,
  useTextAlignProp,
  useKeyOrValueAndKey,
  useKeyOnly,
} from '../../../lib';

import Icon from '../../media/Icon';
import Image from '../../media/Image';

import HeadingSubheading from './HeadingSubheading';
import HeadingContent from './HeadingContent';

/**
 * A heading provides a short summary of content
 */
const Heading = (props) => {
  const {
    attached,
    block,
    children,
    className,
    color,
    content,
    disabled,
    dividing,
    floated,
    icon,
    image,
    inverted,
    size,
    sub,
    subheading,
    textAlign,
  } = props;

  const classes = cx(
    'ui',
    color,
    size,
    useKeyOnly(block, 'block'),
    useKeyOnly(disabled, 'disabled'),
    useKeyOnly(dividing, 'dividing'),
    useValueAndKey(floated, 'floated'),
    useKeyOnly(icon === true, 'icon'),
    useKeyOnly(image === true, 'image'),
    useKeyOnly(inverted, 'inverted'),
    useKeyOnly(sub, 'sub'),
    useKeyOrValueAndKey(attached, 'attached'),
    useTextAlignProp(textAlign),
    'heading',
    className,
  );

  const rest = getUnhandledProps(Heading, props);
  const ElementType = getElementType(Heading, props);

  if (!_.isNil(children)) {
    return <ElementType {...rest} className={classes}>{children}</ElementType>;
  }

  const iconElement = Icon.create(icon);
  const imageElement = Image.create(image);
  const subheadingElement = HeadingSubheading.create(subheading);

  if (iconElement || imageElement) {
    return (
      <ElementType {...rest} className={classes}>
        {iconElement || imageElement}
        {(content || subheadingElement) && (
          <HeadingContent>
            {content}
            {subheadingElement}
          </HeadingContent>
        )}
      </ElementType>
    );
  }

  return (
    <ElementType {...rest} className={classes}>
      {content}
      {subheadingElement}
    </ElementType>
  );
};

Heading._meta = {
  name: 'Heading',
  type: META.TYPES.TEXT,
};

Heading.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Attach heading  to other content, like a segment. */
  attached: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.oneOf(['top', 'bottom']),
  ]),

  /** Format heading to appear inside a content block. */
  block: PropTypes.bool,

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Color of the heading. */
  color: PropTypes.oneOf(UV5UI.COLORS),

  /** Shorthand for primary content. */
  content: customPropTypes.contentShorthand,

  /** Show that the heading is inactive. */
  disabled: PropTypes.bool,

  /** Divide heading from the content below it. */
  dividing: PropTypes.bool,

  /** Heading can sit to the left or right of other content. */
  floated: PropTypes.oneOf(UV5UI.FLOATS),

  /** Add an icon by icon name or pass an Icon. */
  icon: customPropTypes.every([
    customPropTypes.disallow(['image']),
    PropTypes.oneOfType([
      PropTypes.bool,
      customPropTypes.itemShorthand,
    ]),
  ]),

  /** Add an image by img src or pass an Image. */
  image: customPropTypes.every([
    customPropTypes.disallow(['icon']),
    PropTypes.oneOfType([
      PropTypes.bool,
      customPropTypes.itemShorthand,
    ]),
  ]),

  /** Inverts the color of the heading for dark backgrounds. */
  inverted: PropTypes.bool,

  /** Content headings are sized with em and are based on the font-size of their container. */
  size: PropTypes.oneOf(_.without(UV5UI.SIZES, 'big', 'massive')),

  /** Headings may be formatted to label smaller or de-emphasized content. */
  sub: PropTypes.bool,

  /** Shorthand for Heading.Subheading. */
  subheading: customPropTypes.itemShorthand,

  /** Align heading content. */
  textAlign: PropTypes.oneOf(UV5UI.TEXT_ALIGNMENTS),
};

Heading.Content = HeadingContent;
Heading.Subheading = HeadingSubheading;

export default Heading;
