// unchained-ui-react/src/components/text/Heading/HeadingContent.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * Heading content wraps the main content when there is an adjacent Icon or Image.
 */
const HeadingContent = (props) => {
  const {
    children,
    className
  } = props;

  const classes = cx(
    'content',
    className
  );

  const rest = getUnhandledProps(HeadingContent, props);
  const ElementType = getElementType(HeadingContent, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

HeadingContent._meta = {
  name: 'HeadingContent',
  parent: 'Heading',
  type: META.TYPES.TEXT,
};

HeadingContent.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,
};

export default HeadingContent;
