// unchained-ui-react/src/components/text/Heading/HeadingSubheading.jsx
import _ from 'lodash';
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  createShorthandFactory,
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * Headings may contain subheadings.
 */
const HeadingSubheading = (props) => {
  const {
    children,
    className,
    content
  } = props;

  const classes = cx(
    'sub heading',
    className
  );

  const rest = getUnhandledProps(HeadingSubheading, props);
  const ElementType = getElementType(HeadingSubheading, props);

  return (
    <ElementType {...rest} className={classes}>
      {_.isNil(children) ? content : children}
    </ElementType>
  );
};

HeadingSubheading._meta = {
  name: 'HeadingSubheading',
  parent: 'Heading',
  type: META.TYPES.TEXT,
};

HeadingSubheading.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Shorthand for primary content. */
  content: customPropTypes.contentShorthand,
};

HeadingSubheading.create = createShorthandFactory(HeadingSubheading, content => ({ content }));

export default HeadingSubheading;
