// unchained-ui-react/src/lib/META.js
import _ from 'lodash/fp';

export const TYPES = {
  CONTAINER: 'containers',
  CONTROL: 'controls',
  LAYOUT: 'layouts',
  MEDIA: 'media',
  NAVIGATION: 'navigation',
  TEXT: 'text',
};

const TYPE_VALUES = _.values(TYPES);

/**
 * Determine if an object qualifies as a META object.
 * It must have the required keys and valid values.
 * @private
 * @param {Object} _meta A proposed component _meta object.
 * @returns {Boolean}
 */
export const isMeta = (_meta) => (
  _.includes(_.get('type', _meta), TYPE_VALUES)
);

/**
 * Extract a component's _meta object and optional key.
 * Handles literal _meta objects, classes with _meta, objects with _meta
 * @private
 * @param {function|object} metaArg A class, a component instance, or meta object..
 * @returns {object|string|undefined}
 */
const getMeta = (metaArg) => { // eslint-disable-line consistent-return
  // literal
  if (isMeta(metaArg)) {
    return metaArg;
  } else if (isMeta(_.get('_meta', metaArg))) {
    // from prop
    return metaArg._meta;
  } else if (isMeta(_.get('constructor._meta', metaArg))) {
    // from class
    return metaArg.constructor._meta;
  }
};

const metaHasKeyValue = _.curry((key, val, metaArg) => _.flow(getMeta, _.get(key), _.eq(val))(metaArg));
export const isType = metaHasKeyValue('type');

// ----------------------------------------
// Export
// ----------------------------------------

// type
export const isContainer = isType(TYPES.CONTAINER);
export const isControl = isType(TYPES.CONTROL);
export const isLayout = isType(TYPES.LAYOUT);
export const isMedia = isType(TYPES.MEDIA);
export const isNavigation = isType(TYPES.NAVIGATION);
export const isText = isType(TYPES.Text);

// parent
export const isParent = _.flow(getMeta, _.has('parent'), _.eq(false));
export const isChild = _.flow(getMeta, _.has('parent'));

// other
export const isPrivate = _.flow(getMeta, _.get('name'), _.startsWith('_'));
