// unchained-ui-react/src/components/containers/Message/MessageContent.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * A message can contain a content.
 */
const MessageContent = (props) => {
  const {
    children,
    className
  } = props;

  const classes = cx(
    'content',
    className
  );

  const rest = getUnhandledProps(MessageContent, props);
  const ElementType = getElementType(MessageContent, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

MessageContent._meta = {
  name: 'MessageContent',
  parent: 'Message',
  type: META.TYPES.CONTAINER,
};

MessageContent.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,
};

export default MessageContent;
