// unchained-ui-react/src/components/navigation/Menu/MenuHeader.jsx
import cx from 'classnames';
import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * A menu item may include a header or may itself be a header.
 */
const MenuHeader = (props) => {
  const {
    children,
    className,
    content
  } = props;

  const classes = cx(
    'header',
    className
  );

  const rest = getUnhandledProps(MenuHeader, props);
  const ElementType = getElementType(MenuHeader, props);

  return (
    <ElementType {...rest} className={classes}>
      {_.isNil(children) ? content : children}
    </ElementType>
  );
};

MenuHeader._meta = {
  name: 'MenuHeader',
  type: META.TYPES.NAVIGATION,
  parent: 'Menu',
};

MenuHeader.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Shorthand for primary content. */
  content: customPropTypes.contentShorthand,
};

export default MenuHeader;
