// unchained-ui-react/src/components/navigation/Menu/MenuMenu.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * A menu can contain a sub menu.
 */
const MenuMenu = (props) => {
  const {
    children,
    className,
    position
  } = props;

  const classes = cx(
    position,
    'menu',
    className,
  );

  const rest = getUnhandledProps(MenuMenu, props);
  const ElementType = getElementType(MenuMenu, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

MenuMenu._meta = {
  name: 'MenuMenu',
  type: META.TYPES.NAVIGATION,
  parent: 'Menu',
};

MenuMenu.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** A sub menu can take right position. */
  position: PropTypes.oneOf(['right']),
};

export default MenuMenu;
