// unchained-ui-react/src/components/layout/Sidebar/SidebarPusher.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
  useKeyOnly,
} from '../../../lib';

/**
 * A pushable sub-component for Sidebar.
 */
const SidebarPusher = (props) => {
  const {
    className,
    dimmed,
    children
  } = props;

  const classes = cx(
    'pusher',
    useKeyOnly(dimmed, 'dimmed'),
    className,
  );

  const rest = getUnhandledProps(SidebarPusher, props);
  const ElementType = getElementType(SidebarPusher, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

SidebarPusher._meta = {
  name: 'SidebarPusher',
  type: META.TYPES.LAYOUT,
  parent: 'Sidebar',
};

SidebarPusher.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Controls whether or not the dim is displayed. */
  dimmed: PropTypes.bool,
};

export default SidebarPusher;
