// unchained-ui-react/src/components/layout/Sidebar/SidebarPushable.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * A pushable sub-component for Sidebar.
 */
const SidebarPushable = (props) => {
  const {
    className,
    children
  } = props;

  const classes = cx(
    'pushable',
    className
  );

  const rest = getUnhandledProps(SidebarPushable, props);
  const ElementType = getElementType(SidebarPushable, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

SidebarPushable._meta = {
  name: 'SidebarPushable',
  type: META.TYPES.LAYOUT,
  parent: 'Sidebar',
};

SidebarPushable.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,
};

export default SidebarPushable;
