// unchained-ui-react/docs/app/components/organisms/ComponentDoc/index.jsx
import cx from 'classnames';
import _ from 'lodash';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DocumentTitle from 'react-document-title';
// import { Link } from 'react-router-dom';

// import { META } from 'src/lib';

import * as unchainedUIReact from 'src/components'; // eslint-disable-line import/no-duplicates

import {
  Divider,
  Grid,
  Heading,
  Icon,
  List,
} from 'src/components'; // eslint-disable-line import/no-duplicates

import { repoURL } from 'docs/app/utils';

import {
  ComponentProps,
  ComponentExamples,
} from 'docs/app/components';

import docgenInfo from '../../../../docgenInfo.json';

const docgenPaths = _.keys(docgenInfo);

const getDocgenPath = (componentName) => _.find(docgenPaths, path => {
  return new RegExp(`/${componentName}.jsx$`).test(path);
});

const getGithubReactSourceUrl = (componentName) => {
  return `${repoURL}/blob/master/${getDocgenPath(componentName)}`;
};

const linkListStyle = {
  position: 'absolute',
  padding: '0.5em',
  margin: '0.5em',
  top: '0',
  right: '0',
  boxShadow: '0 0 1em 0.5em #f7f7f7',
  background: '#f7f7f7',
};

const showPropsStyle = {
  display: 'inline-flex',
  margin: '1em 0.5em',
  marginLeft: 0,
  cursor: 'pointer',
};

const subDescriptionStyle = {
  fontSize: '1.08em',
  color: '#777',
};

export default class ComponentDoc extends Component {
  static propTypes = {
    _meta: PropTypes.object,
  };

  state = {
    showPropsFor: '',
  };

  componentWillMount() {
    this.setState({
      showPropsFor: this.props._meta.name,
    });
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      showPropsFor: this.state.showPropsFor ? nextProps._meta.name : null,
    });
  }

  toggleProps = (name) => {
    this.setState({
      showPropsFor: this.state.showPropsFor === name ? null : name,
    });
  };

  renderHeader = () => {
    const {
      _meta
    } = this.props;

    const docgen = docgenInfo[getDocgenPath(_meta.name)];

    return (
      <Heading as={'h1'} style={{ marginBottom: '0.25em' }}>
        {_meta.name}
        <Heading.Subheading>
          {docgen.description}
        </Heading.Subheading>
      </Heading>
    );
  };

  renderGithubReactSourceLink() {
    const {
      _meta
    } = this.props;

    const docPath = getDocgenPath(_meta.name);

    return (
      <List.Item>
        <List.Icon name={'github'} />
        <List.Content>
          <code>
            <a href={getGithubReactSourceUrl(_meta.name)} target={'_blank'} rel={'noopener noreferrer'}>{docPath}</a>
          </code>
        </List.Content>
      </List.Item>
    );
  }

  renderGithubScssSourceLink() {
    const {
      _meta
    } = this.props;

    const scssDefPath = getDocgenPath(_meta.name)
      .replace('/components/', '/styles/')
      .replace(`/${_meta.name}.jsx`, `/${_meta.name}.scss`)
      .replace(`/${_meta.name}/`, '/')
      .toLowerCase();

    const githubUrl = `${repoURL}/blob/master/${scssDefPath}`;

    return (
      <List.Item>
        <List.Icon name={'github'} />
        <List.Content>
          <code>
            <a href={githubUrl} target={'_blank'} rel={'noopener noreferrer'}>{scssDefPath}</a>
          </code>
        </List.Content>
      </List.Item>
    );
  }

  renderGithubScssThemeLink() {
    const {
      _meta
    } = this.props;

    const scssThemePath = getDocgenPath(_meta.name)
      .replace('/components/', '/styles/themes/default/')
      .replace(`/${_meta.name}.jsx`, `/${_meta.name}.scss`)
      .replace(`/${_meta.name}/`, '/')
      .toLowerCase();

    const githubUrl = `${repoURL}/blob/master/${scssThemePath}`;

    return (
      <List.Item>
        <List.Icon name={'github'} />
        <List.Content>
          <code>
            <a href={githubUrl} target={'_blank'} rel={'noopener noreferrer'}>{scssThemePath}</a>
          </code>
        </List.Content>
      </List.Item>
    );
  }

  renderProps = () => {
    const {
      _meta
    } = this.props;

    const {
      showPropsFor
    } = this.state;

    const itemCX = name => cx(
      showPropsFor === name && 'active',
      'link item'
    );

    const selectedDocgen = docgenInfo[getDocgenPath(showPropsFor)];
    const toggleIcon = `toggle ${showPropsFor ? 'on' : 'off'}`;

    const subComponents = _.flatMap(unchainedUIReact, UV5Component => _.filter(UV5Component, staticValue => (
      _.get(staticValue, '_meta.parent') === _meta.name
    )));

    const hasSubComponents = !_.isEmpty(subComponents);
    const showSubDescription = hasSubComponents && _.get(selectedDocgen, 'description');

    /* eslint-disable jsx-a11y/no-static-element-interactions */
    const subComponentItems = _.map(subComponents, ({ _meta: { name } }) => (
      <div key={name} className={itemCX(name)} onClick={() => this.toggleProps(name)}>
        {_meta.name}.{name.replace(_meta.name, '')}
      </div>
    ));

    return (
      <div>
        <Heading
          as={'h4'}
          className={'no-anchor'}
          color={cx(showPropsFor ? 'green' : 'grey')}
          style={showPropsStyle}
          onClick={() => this.toggleProps(showPropsFor || _meta.name)}
        >
          <a style={{ color: 'inherit' }}><Icon name={toggleIcon} /> Props{hasSubComponents && ':'}</a>
        </Heading>
        <div className={'ui small compact secondary green menu'}>
          {hasSubComponents && (
            <div className={itemCX(_meta.name)} onClick={() => this.toggleProps(_meta.name)}>
              {_meta.name}
            </div>
          )}
          {subComponentItems}
        </div>
        {selectedDocgen && (
          <div>
            {showSubDescription && (
              <div style={subDescriptionStyle}>
                {selectedDocgen.description}
                <Divider />
              </div>
            )}
            <ComponentProps props={selectedDocgen.props} meta={unchainedUIReact[showPropsFor]._meta} />
          </div>
        )}
      </div>
    );
    /* eslint-enable jsx-a11y/no-static-element-interactions */
  };

  render() {
    const {
      _meta
    } = this.props;

    return (
      <DocumentTitle title={`${_meta.name} | Unchained UI`}>
        <div>
          <Grid padded columns={'1'}>
            <Grid.Column>
              {this.renderHeader()}

              <List link style={linkListStyle}>
                {this.renderGithubReactSourceLink()}
                {this.renderGithubScssSourceLink()}
                {this.renderGithubScssThemeLink()}
              </List>

              {this.renderProps()}
            </Grid.Column>
          </Grid>

          <ComponentExamples name={_meta.name} />
        </div>
      </DocumentTitle>
    );
  }
}
