// unchained-ui-react/docs/app/components/molecules/AppSidebar/index.jsx
import _ from 'lodash/fp';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { findDOMNode } from 'react-dom';
import { NavLink } from 'react-router-dom';
import { withRouter } from 'react-router';

import pkg from 'package.json';

import {
  keyboardKey,
  META,
} from 'src/lib';

import {
  parentComponents,
  typeOrder,
} from 'docs/app/utils';

import {
  Icon,
  Input,
  Menu,
  Sidebar,
} from 'src/components';

const getRoute = (_meta) => `/${_meta.type}/${_.kebabCase(_meta.name)}`;
const selectedItemLabel = <span style={{ color: '#35bdb2', float: 'right' }}>Press Enter</span>;

class AppSidebar extends Component {
  static propTypes = {
    history: PropTypes.object.isRequired,
    sidebarVisible: PropTypes.bool.isRequired,
    toggleSidebar: PropTypes.func.isRequired,
  };

  state = {
    query: ''
  };

  componentDidMount() {
    document.addEventListener('keydown', this.handleDocumentKeyDown);
    this.setSearchInput();
  }

  componentDidUpdate(prevProps, prevState) { // eslint-disable-line no-unused-vars-rest/no-unused-vars
    this.setSearchInput();
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleDocumentKeyDown);
  }

  setSearchInput() {
    this._searchInput = findDOMNode(this).querySelector('.ui.input input'); // eslint-disable-line react/no-find-dom-node
  }

  filteredComponents = parentComponents;

  handleDocumentKeyDown = (e) => {
    const code = keyboardKey.getCode(e);
    const isAZ = code >= 65 && code <= 90;
    const hasModifier = e.altKey || e.ctrlKey || e.metaKey;
    const bodyHasFocus = document.activeElement === document.body;

    if (!hasModifier && isAZ && bodyHasFocus) this._searchInput.focus();
  }

  handleItemClick = () => {
    const {
      query
    } = this.state;

    if (query) {
      this.setState({
        query: '',
      });
    }

    if (document.activeElement === this._searchInput) this._searchInput.blur();

    this.props.toggleSidebar();
  };

  handleSearchChange = e => this.setState({
    selectedItemIndex: 0,
    query: e.target.value,
  });

  handleSearchKeyDown = e => {
    const {
      history
    } = this.props;

    const {
      selectedItemIndex
    } = this.state;

    const code = keyboardKey.getCode(e);

    if (code === keyboardKey.Enter && this.selectedRoute) {
      e.preventDefault();
      history.push(this.selectedRoute);
      this.selectedRoute = null;
      this._searchInput.blur();

      this.setState({
        query: '',
      });

      this.props.toggleSidebar();
    }

    if (code === keyboardKey.ArrowDown) {
      e.preventDefault();

      const next = _.min([selectedItemIndex + 1, this.filteredComponents.length - 1]);

      this.selectedRoute = getRoute(this.filteredComponents[next]._meta);

      this.setState({
        selectedItemIndex: next,
      });
    }

    if (code === keyboardKey.ArrowUp) {
      e.preventDefault();

      const next = _.max([selectedItemIndex - 1, 0]);

      this.selectedRoute = getRoute(this.filteredComponents[next]._meta);

      this.setState({
        selectedItemIndex: next,
      });
    }
  }

  menuItemsByType = _.map((type) => {
    const items = _.flow(
      _.filter(META.isType(type)),
      _.map(({ _meta }) => (
        <Menu.Item
          key={_meta.name}
          name={_meta.name}
          onClick={this.handleItemClick}
          as={NavLink}
          to={getRoute(_meta)}
          activeClassName={'active'}
        />
      ))
    )(parentComponents);

    return (
      <Menu.Item key={type}>
        <Menu.Header>{_.capitalize(type)}</Menu.Header>
        <Menu.Menu>{items}</Menu.Menu>
      </Menu.Item>
    );
  }, typeOrder);

  renderSearchItems = () => {
    const {
      selectedItemIndex,
      query
    } = this.state;

    if (!query) return;

    let itemIndex = -1;
    const startsWithMatches = [];
    const containsMatches = [];

    _.each(component => {
      if (component._meta) {
        if (new RegExp(`^${_.escapeRegExp(query)}`, 'i').test(component._meta.name)) {
          startsWithMatches.push(component);
        } else if (new RegExp(_.escapeRegExp(query), 'i').test(component._meta.name)) {
          containsMatches.push(component);
        }
      }
    }, parentComponents);

    this.filteredComponents = [
      ...startsWithMatches,
      ...containsMatches
    ];

    const menuItems = _.map(({ _meta }) => {
      itemIndex += 1;

      const isSelected = itemIndex === selectedItemIndex;

      if (isSelected) this.selectedRoute = getRoute(_meta);

      return (
        <Menu.Item
          key={_meta.name}
          name={_meta.name}
          onClick={this.handleItemClick}
          active={isSelected}
          as={NavLink}
          to={getRoute(_meta)}
        >
          {_meta.name}
          {isSelected && selectedItemLabel}
        </Menu.Item>
      );
    }, this.filteredComponents);

    return <Menu.Menu>{menuItems}</Menu.Menu>; // eslint-disable-line consistent-return
  };

  render() {
    const {
      sidebarVisible,
      toggleSidebar,
    } = this.props;

    const {
      query
    } = this.state;

    return (
      <Sidebar animation={'overlay'} as={Menu} visible={sidebarVisible} vertical inverted size={'large'}>
        <Menu.Item>
          <strong>
            Unchained UI &nbsp;
            <small><em>{pkg.version}</em></small>
          </strong>
          <Icon name={'close'} onClick={toggleSidebar} link />
        </Menu.Item>
        <Menu.Item>
          <Menu.Header>Getting Started</Menu.Header>
          <Menu.Menu>
            <Menu.Item as={NavLink} to={'/intro'} activeClassName={'active'}>
              Introduction
            </Menu.Item>
          </Menu.Menu>
        </Menu.Item>
        <Menu.Item>
          <Input
            className={'transparent inverted icon'}
            icon={'search'}
            placeholder={'Start typing...'}
            value={query}
            onChange={this.handleSearchChange}
            onKeyDown={this.handleSearchKeyDown}
          />
        </Menu.Item>
        {query ? this.renderSearchItems() : this.menuItemsByType}
      </Sidebar>
    );
  }
}

export default withRouter(AppSidebar);
