// unchained-ui-react/src/components/controls/Button/ButtonContent.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
  useKeyOnly,
} from '../../../lib';

/**
 * Used in some Button types, such as `animated`.
 */
const ButtonContent = (props) => {
  const {
    children,
    className,
    hidden,
    visible,
  } = props;

  const classes = cx(
    useKeyOnly(visible, 'visible'),
    useKeyOnly(hidden, 'hidden'),
    'content',
    className,
  );

  const rest = getUnhandledProps(ButtonContent, props);
  const ElementType = getElementType(ButtonContent, props);

  return <ElementType {...rest} className={classes}>{children}</ElementType>;
};

ButtonContent._meta = {
  name: 'ButtonContent',
  parent: 'Button',
  type: META.TYPES.CONTROL,
};

ButtonContent.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Initially hidden, visible on hover. */
  hidden: PropTypes.bool,

  /** Initially visible, hidden on hover. */
  visible: PropTypes.bool,
};

export default ButtonContent;
