// unchained-ui-react/src/components/controls/Button/ButtonOr.jsx
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * Button groups can contain conditionals.
 */
const ButtonOr = (props) => {
  const {
    className,
    text
  } = props;

  const classes = cx(
    'or',
    className
  );

  const rest = getUnhandledProps(ButtonOr, props);
  const ElementType = getElementType(ButtonOr, props);

  return <ElementType {...rest} className={classes} data-text={text} />;
};

ButtonOr._meta = {
  name: 'ButtonOr',
  parent: 'Button',
  type: META.TYPES.CONTROL,
};

ButtonOr.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Additional classes. */
  className: PropTypes.string,

  /** Or buttons can have their text localized, or adjusted by using the text prop. */
  text: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
};

export default ButtonOr;
