// unchained-ui-react/src/components/containers/Card/CardMeta.jsx
import cx from 'classnames';
import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

/**
 * A card can contain content metadata.
 */
const CardMeta = (props) => {
  const {
    children,
    className,
    content
  } = props;

  const classes = cx(
    className,
    'meta'
  );

  const rest = getUnhandledProps(CardMeta, props);
  const ElementType = getElementType(CardMeta, props);

  return (
    <ElementType {...rest} className={classes}>
      {_.isNil(children) ? content : children}
    </ElementType>
  );
};

CardMeta._meta = {
  name: 'CardMeta',
  parent: 'Card',
  type: META.TYPES.CONTAINER,
};

CardMeta.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Shorthand for primary content. */
  content: customPropTypes.contentShorthand,
};

export default CardMeta;
