// unchained-ui-react/docs/app/components/molecules/ComponentProps/index.jsx
import _ from 'lodash';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {
  UV5UI,
} from 'src/lib';

import {
  Icon,
  Popup,
  Table,
} from 'src/components';

import {
  Extra,
} from 'docs/app/components';

const popupIcon = <Icon size={'small'} color={'red'} name={'asterisk'} />;
const getTagType = tag => tag.type.type === 'AllLiteral' ? 'any' : tag.type.name;

export default class ComponentProps extends Component {
  static propTypes = {
    /**
     * A single Component's prop info as generated by react-docgen.
     * @type {object} Props info object where keys are prop names and values are prop definitions.
     */
    props: PropTypes.object,
  };

  state = {
    showEnumsFor: {},
  };

  expandEnums = (value) => {
    const parts = value.split('.');

    if (parts[0] === 'UV5UI') {
      return UV5UI[parts[1]];
    }

    return value;
  };

  toggleEnumsFor = (prop) => () => {
    this.setState({
      showEnumsFor: {
        ...this.state.showEnumsFor,
        [prop]: !this.state.showEnumsFor[prop],
      },
    });
  };

  renderRequired = item => item.required && (
    <Popup
      position={'right center'}
      style={{ padding: '0.5em' }}
      trigger={popupIcon}
      content={'Required'}
      size={'tiny'}
      inverted
    />
  );

  renderDefaultValue = item => {
    const defaultValue = _.get(item, 'defaultValue.value');

    if (_.isNil(defaultValue)) return null;

    return <code>{defaultValue}</code>;
  };

  renderFunctionSignature = (item) => {
    const params = _.filter(item.tags, { title: 'param' });
    const returns = _.find(item.tags, { title: 'returns' });

    // this doesn't look like a function propType doc block
    // don't try to render a signature
    if (_.isEmpty(params) && !returns) return;

    const paramSignature = params
      .map(param => `${param.name}: ${getTagType(param)}`)
      // prevent object properties from showing as individual params
      .filter(p => !p.includes('.'))
      .join(', ');

    const tagDescriptionRows = _.compact([...params, returns]).map(tag => {
      const name = tag.name || tag.title;

      return (
        <div key={name} style={{ display: 'flex', flexDirection: 'row' }}>
          <div style={{ flex: '2 2 0', padding: '0.1em 0' }}>
            <code>{name}</code>
          </div>
          <div style={{ flex: '5 5 0', padding: '0.1em 0' }}>
            {tag.description}
          </div>
        </div>
      );
    });

    return ( // eslint-disable-line consistent-return
      <Extra title={<pre>{item.name}({paramSignature}){returns ? `: ${getTagType(returns)}` : ''}</pre>}>
        {tagDescriptionRows}
      </Extra>
    );
  };

  renderName = item => <code>{item.name}</code>;

  renderEnums = (item) => {
    if (item.type !== '{enum}') return;

    const {
      showEnumsFor
    } = this.state;

    const truncateAt = 10;

    if (!item.value) return null; // eslint-disable-line consistent-return

    const values = [].concat(item.value).reduce((accumulator, v) => {
      return accumulator.concat(this.expandEnums(_.trim(v.value || v, '.\'')));
    }, []);

    const valueElements = _.map(values, val => <span key={val}><code>{`${val} `}</code> </span>);

    // show all if there are few
    if (values.length < truncateAt) {
      return ( // eslint-disable-line consistent-return
        <Extra title={'Enums:'} inline>
          {valueElements}
        </Extra>
      );
    }

    // add button to show more when there are many values and it is not toggled
    if (!showEnumsFor[item.name]) {
      /* eslint-disable jsx-a11y/no-static-element-interactions */
      return ( // eslint-disable-line consistent-return
        <Extra title={'Enums:'} inline>
          <a style={{ cursor: 'pointer' }} onClick={this.toggleEnumsFor(item.name)}>
            Show all {values.length}
          </a>
          <div>{valueElements.slice(0, truncateAt - 1)}...</div>
        </Extra>
      );
      /* eslint-enable jsx-a11y/no-static-element-interactions */
    }

    /* eslint-disable jsx-a11y/no-static-element-interactions */
    // add "show more" button when there are many
    return ( // eslint-disable-line consistent-return
      <Extra title={'Enums:'} inline>
        <a style={{ cursor: 'pointer' }} onClick={this.toggleEnumsFor(item.name)}>
          Show less
        </a>
        <div>{valueElements}</div>
      </Extra>
    );
    /* eslint-enable jsx-a11y/no-static-element-interactions */
  };

  renderRow = item => {
    return (
      <Table.Row key={item.name}>
        <Table.Cell>{this.renderName(item)}{this.renderRequired(item)}</Table.Cell>
        <Table.Cell>{this.renderDefaultValue(item)}</Table.Cell>
        <Table.Cell>{item.type}</Table.Cell>
        <Table.Cell>
          {item.description && <p>{item.description}</p>}
          {this.renderFunctionSignature(item)}
          {this.renderEnums(item)}
        </Table.Cell>
      </Table.Row>
    );
  };

  render() {
    const {
      props: propsDefinition
    } = this.props;

    const content = _.sortBy(_.map(propsDefinition, (config, name) => {
      const value = _.get(config, 'type.value');
      let type = _.get(config, 'type.name');

      if (type === 'union') {
        type = _.map(value, (val) => val.name).join('|');
      }

      type = type && `{${type}}`;

      const description = _.get(config, 'description', '');

      return {
        name,
        type,
        value,
        // tags: _.get(config, 'docBlock.tags'),
        required: config.required,
        defaultValue: config.defaultValue,
        description: description && description.split('\n').map(l => ([l, <br key={l} />])),
      };
    }), 'name');

    return (
      <Table compact={'very'} basic={'very'}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Name</Table.HeaderCell>
            <Table.HeaderCell>Default</Table.HeaderCell>
            <Table.HeaderCell>Type</Table.HeaderCell>
            <Table.HeaderCell>Description</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {_.map(content, this.renderRow)}
        </Table.Body>
      </Table>
    );
  }
}
