// unchained-ui-react/src/components/media/Label/LabelDetail.jsx
import cx from 'classnames';
import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import {
  customPropTypes,
  getElementType,
  getUnhandledProps,
  META,
} from '../../../lib';

const LabelDetail = (props) => {
  const {
    children,
    className,
    content
  } = props;

  const classes = cx(
    'detail',
    className
  );

  const rest = getUnhandledProps(LabelDetail, props);
  const ElementType = getElementType(LabelDetail, props);

  return (
    <ElementType {...rest} className={classes}>
      {_.isNil(children) ? content : children}
    </ElementType>
  );
};

LabelDetail._meta = {
  name: 'LabelDetail',
  parent: 'Label',
  type: META.TYPES.MEDIA,
};

LabelDetail.propTypes = {
  /** An element type to render as (string or function). */
  as: customPropTypes.as, // eslint-disable-line react/no-unused-prop-types

  /** Primary content. */
  children: PropTypes.node,

  /** Additional classes. */
  className: PropTypes.string,

  /** Shorthand for primary content. */
  content: customPropTypes.contentShorthand,
};

export default LabelDetail;
